src/components/core/GeoChart.tsx

```diff
@@ -40,28 +52,49 @@ export type IGeoChartInnerProps = ICoreGeoChartProps &
     IDataSourceProviderInjectedProps &
     WrappedComponentProps;
 
+export interface IGeoChartInnerState {
+    legendIndexEnabled: boolean[];
+    legendNamesEnabled: string[];
+}
+export interface IGeoChartInnerOptions {
+    geoData: IGeoData;
+    series: any;
+    sizeValues: number[];
+}
 /**
  * Geo Chart react component
  */
-export class GeoChartInner extends BaseVisualization<IGeoChartInnerProps, {}> {
+export class GeoChartInner extends BaseVisualization<IGeoChartInnerProps, IGeoChartInnerState> {
     public static defaultProps: Partial<IGeoChartInnerProps> = {
         ...commonDefaultProps,
         chartRenderer: renderChart,
         legendRenderer: renderLegend,
-        legendPostion: TOP,
+        legendPosition: TOP,
     };
+    private geoChartOptions: IGeoChartInnerOptions;
+    constructor(props: IGeoChartInnerProps) {
+        super(props);
+        this.state = {
+            legendIndexEnabled: [],
+            legendNamesEnabled: [],
+        };
+        this.onLegendItemClick = this.onLegendItemClick.bind(this);
+    }
 
     public componentDidUpdate(prevProps: IGeoChartInnerProps) {
+        console.info("update", this.props.execution); //tslint:disable-line
         if (!isEqual(this.props.execution, prevProps.execution)) {
-            this.validateData();
+            this.transformationData();
         }
```

Comment:
we should separate ` validateData ` logic from `transformData` logic
it could be
```
if (isEqual(this.props.execution, prevProps.execution)) { return; }
if(!this.validateData()) { return; }
this.transformData();
```

Reply:
validateData function have re-use some variables (ex: geoData), so it can't separate  function.

Reply:
fixed